### PR TITLE
[IOTDB-2159] Fix wrong status code when inserting data to a read-only cluster

### PR DESCRIPTION
**Pull request by ericpai**
See JIRA: https://issues.apache.org/jira/browse/IOTDB-2159

### CODE REVIEW

**Comment by ericpai** [comment]
@LebronAl PTAL

File: cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java [hunk 1]
```diff
@@ -71,7 +71,8 @@ protected TSStatus executeNonQueryPlan(PhysicalPlan plan) {
       if (!(plan instanceof SetSystemModePlan)
           && !(plan instanceof FlushPlan)
           && IoTDBDescriptor.getInstance().getConfig().isReadOnly()) {
-        throw new QueryProcessException(
+        return RpcUtils.getStatus(
+            TSStatusCode.QUERY_PROCESS_ERROR,
             "Current system mode is read-only, does not support non-query operation");
```
Comment (HTHou) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: I just feel a little bit weird why a NonQueryPlan throws a  QUERY_PROCESS_ERROR...
Should we change it to another error code?

Comment (ericpai) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: > I just feel a little bit weird why a NonQueryPlan throws a QUERY_PROCESS_ERROR... Should we change it to another error code?

It makes sense. Is `WRITE_PROCESS_REJECT(413)` OK?

Comment (HTHou) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: How about the `READ_ONLY_SYSTEM_ERROR(502)`?

Comment (ericpai) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: > How about the `READ_ONLY_SYSTEM_ERROR(502)`?

Althogh the name is reasonable, the description is not as our expected.
image

Comment (HTHou) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: > Althogh the name is reasonable, the description is not as our expected. image

Is there any problem about the description?

Comment (ericpai) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).

Comment (HTHou) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java:76]: > I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).

Makes sense. Yes, `database is read only` looks good to me.

Comment (ericpai) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/ClusterTSServiceImpl.java]: > > I haven't found any usage of READ_ONLY_SYSTEM_ERROR yet. Maybe we can change the sentence `Operating system is read only` to `database is read only`? The 'Opertating system' is confused with our common word `OS`(Unix or Windows).
> 
> Makes sense. Yes, `database is read only` looks good to me.

I refine the codes and add a new exception class. Please take a review again.

**Review by LebronAl — APPROVED** [review]
LGTM

**Review by HTHou — APPROVED** [review]

**Pull request merged**